### PR TITLE
Add d2l lti activity component

### PR DESCRIPTION
**Pull request by alogan519**
Add skeleton support to hc-description

### CODE REVIEW

File: features/lti/README.MD [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+**Properties:**
+
+| Property | Type | Description |
+|--|--|--|
+| 'open-as-external' | Boolean | Only Render a button to open in a new window |
+
+### Example
+
+```html
+<d2l-lti-activity href="${activity usage href}" .token="${token}"></d2l-discover-rules>
```
Comment (maggiesazio) [inline features/lti/README.MD:18]:
```suggestion
<d2l-lti-activity href="${activity usage href}" .token="${token}"></d2l-lti-activity>

```

File: features/lti/README.MD [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+<d2l-lti-activity href="${activity usage href}" .token="${token}" open-as-external></d2l-discover-rules>
```
Comment (maggiesazio) [inline features/lti/README.MD:19]:
```suggestion
<d2l-lti-activity href="${activity usage href}" .token="${token}" open-as-external></d2l-lti-activity>

```

File: features/lti/d2l-lti-activity.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+			</div>
+			<div class="d2l-body-compact">
+				${this.localize('external-activity-check-below')}
+			</div>
+
+			${this.skeleton ? html`<div class="d2l-skeletize skeleton-placeholder"></div>` :	
+
+				html`${this._showOpenInNewWindowButton ? 
+					html`<d2l-button class="spanning-button" primary @click="${this._onOpenInNewWindowClick}">${this.localize('open-in-new-window')}</d2l-button>` :
+					html`<iframe width="${this.iFrameWidth}" height="${this.iFrameHeight}" src="${this.launchUrl}"></iframe>
```
Comment (maggiesazio) [inline features/lti/d2l-lti-activity.js:108]: I wonder if we should include `allow=\"microphone *; camera *; autoplay *` for the iframe. We have that for LTI iframes in  files and in content.

Comment (alogan519) [reply features/lti/d2l-lti-activity.js:108]: ok yes thank you, I will add that in.

File: components/common/d2l-hc-description.js [hunk 4]
```diff
@@ -2,21 +2,40 @@ import { css, LitElement } from 'lit-element/lit-element.js';
 import { HypermediaStateMixin, observableTypes} from '@brightspace-hmc/foundation-engine/framework/lit/HypermediaStateMixin.js';
 import { html } from '@brightspace-hmc/foundation-engine/framework/lit/hypermedia-components.js';
 
-class HmDescription extends HypermediaStateMixin(LitElement) {
+import { SkeletonMixin } from '@brightspace-ui/core/components/skeleton/skeleton-mixin.js';
+
+class HmDescription extends SkeletonMixin(HypermediaStateMixin(LitElement)) {
 	static get properties() {
 		return {
 			description: { type: String, observable: observableTypes.property }
```
Comment (m6jones) [inline components/common/d2l-hc-description.js:10]: Description is private so it should be `_description` I made it so the engine will filter out the `_` so you can still use the short form.

Another thing you could do is extend descriptionMixin which is in components->activities->description->custom
Then you just have to set this property.

**Review by m6jones — APPROVED** [review]

File: components/common/d2l-hc-description.js (hunk 4 above)
Comment (alogan519) [reply components/common/d2l-hc-description.js:10]: Ok thanks, I'll go with the _description

**Pull request merged**